application/src/test/java/org/thingsboard/server/service/apiusage/ApiUsageTest.java

```diff
@@ -0,0 +1,144 @@
+import org.thingsboard.server.controller.AbstractControllerTest;
+import org.thingsboard.server.controller.TbUrlConstants;
+import org.thingsboard.server.dao.service.DaoSqlTest;
+import org.thingsboard.server.dao.usagerecord.ApiUsageStateService;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@DaoSqlTest
+@TestPropertySource(properties = {
+        "usage.stats.report.enabled=true",
+        "usage.stats.report.interval=2",
+        "usage.stats.gauge_report_interval=1",
+})
+public class ApiUsageTest extends AbstractControllerTest {
+
+    private Tenant savedTenant;
+    private User tenantAdmin;
+
+    private static final int MAX_DP_ENABLE_VALUE = 12;
+    private static final double WARN_THRESHOLD_VALUE = 0.5;
+    @Autowired
+    private ApiUsageStateService apiUsageStateService;
+
+    @Before
+    public void beforeTest() throws Exception {
+        loginSysAdmin();
+
+        TenantProfile tenantProfile = createTenantProfile();
+        TenantProfile savedTenantProfile = doPost("/api/tenantProfile", tenantProfile, TenantProfile.class);
+        assertNotNull(savedTenantProfile);
+
+        Tenant tenant = new Tenant();
+        tenant.setTitle("My tenant");
+        tenant.setTenantProfileId(savedTenantProfile.getId());
+        savedTenant = saveTenant(tenant);
+        tenantId = savedTenant.getId();
+        assertNotNull(savedTenant);
+
+        tenantAdmin = new User();
+        tenantAdmin.setAuthority(Authority.TENANT_ADMIN);
+        tenantAdmin.setTenantId(savedTenant.getId());
+        tenantAdmin.setEmail("tenant2@thingsboard.org");
+
+        tenantAdmin = createUserAndLogin(tenantAdmin, "testPassword1");
+    }
+
+    @Test
+    public void testTelemetryApiCall() throws Exception {
+        Device device = createDevice();
+        assertNotNull(device);
+        String telemetryPayload = "{\"temperature\":25, \"humidity\":60}";
+        String url = TbUrlConstants.TELEMETRY_URL_PREFIX + "/DEVICE/" + device.getId() + "/timeseries/ANY";
+
+        long VALUE_WARNING = (long) (MAX_DP_ENABLE_VALUE * WARN_THRESHOLD_VALUE) / 2;
+
+        for (int i = 0; i < VALUE_WARNING; i++) {
+            doPostAsync(url, telemetryPayload, String.class, status().isOk());
+        }
+
+        await().atMost(5, TimeUnit.SECONDS).untilAsserted(() -> assertEquals(ApiUsageStateValue.WARNING, apiUsageStateService.findTenantApiUsageState(tenantId).getDbStorageState()));
+
+        long VALUE_DISABLE = (long) (MAX_DP_ENABLE_VALUE - (MAX_DP_ENABLE_VALUE * WARN_THRESHOLD_VALUE)) / 2;
+
+        for (int i = 0; i < VALUE_DISABLE; i++) {
+            doPostAsync(url, telemetryPayload, String.class, status().isOk());
+        }
+
+        await().atMost(5, TimeUnit.SECONDS)
```

Comment:
5 seconds is too little, the test is going to be flaky on a build machine, use TIMEOUT constant (30 seconds)